### PR TITLE
Add opportunity provider script

### PR DESCRIPTION
**Pull request by danimhr**
This PR aims to provide a script for the opportunity provider

### CODE REVIEW

File: scripts/opportunity-provider/package.json [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+  "main": "dist/index.js",
+  "types": "dist/index.d.ts",
+  "scripts": {
+    "build": "tsc",
+    "test": "echo \"Error: no test specified\" && exit 1",
+    "provide-opportunity": "npm run build && node dist/index.js"
+  },
+  "keywords": [],
+  "author": "",
+  "license": "ISC",
```
Comment (m30m) [inline scripts/opportunity-provider/package.json:14]: Apache

File: scripts/opportunity-provider/index.ts [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+  console.log(data);
+}
+
+const argv = yargs(hideBin(process.argv))
+  .option("private-key", {
+    description:
+      "Private key to provide opportunity in hex format with 0x prefix. e.g: 0xdeadbeef...",
+    type: "string",
+    demandOption: true,
+  })
```
Comment (m30m) [inline scripts/opportunity-provider/index.ts:208]: maybe add options for config and opportunity path with default values set to `config.json` and `opportunity.json`. I can imagine having multiple pre-built opportunities for testing different scenarios and I'd rather not change the same file over and over again.

**Review by m30m — APPROVED** [review]

File: scripts/opportunity-provider/index.ts (hunk 2 above)
Comment (anihamde) [reply scripts/opportunity-provider/index.ts:208]: could also add multiple opportunities to `opportunity.json`. i guess it depends in part on whether we want to create the same set of opportunities again and again. if so, then i'd say just add to a single json.

File: contracts/script/Vault.s.sol [hunk 3]
```diff
@@ -149,6 +150,24 @@ contract VaultScript is Script {
...
+
+        OpportunityProvider provider = new OpportunityProvider(
+            admin,
+            expressRelay,
+            permit2
+        );
+        vm.stopBroadcast();
+        console.log("Provoder at", address(provider));
+        return address(provider);
+    }
```
[the remark this answers is not in the document]
Comment (danimhr) [reply contracts/script/Vault.s.sol:169]: Where do we use deployAll?

**Pull request merged**